File: Java/src/main/java/de/keineInsektenImEssen/textdetector/model/Synonym.java
package de.keineInsektenImEssen.textdetector.model;

import java.util.List;

// FK-TODO: convert to record class
public class Synonym {

    public final String name;
    public final List<Pattern> patterns;

    public Synonym(final String name, final List<Pattern> patterns) {
        this.name = name;
        this.patterns = patterns;
    }

    @Override
    public String toString() {
        return "Synonym{" +
                "name='" + name + '\'' +
                ", patterns=" + patterns +
                '}';
    }
}

File: docs/InsectsInFoodChecker/Synonym.js
class Synonym {

    name;
    patterns;
    url;

    constructor(name, patterns, url) {
        this.name = name;
        this.patterns = patterns;
        this.url = url;
    }

    findAllMatches(haystack) {
        return Patterns.findAllMatches(this.patterns, haystack);
    }
}
